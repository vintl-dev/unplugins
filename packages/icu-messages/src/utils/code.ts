/**
 * Checks whether the provided code has singns of being already transformed.
 *
 * @param code The code to check.
 * @returns Whether the code has already been transformed.
 */
export function isTransformed(code_: string) {
  const code = code_.trim()

  for (const badToken of ['const', 'let', 'var', 'export', 'import']) {
    if (code.startsWith(badToken)) return true
  }

  return false
}
