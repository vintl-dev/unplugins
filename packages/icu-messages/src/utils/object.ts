/**
 * Accepts an object as input and checks whether the object is `undefined`,
 * `null`, or does not has own properties.
 *
 * @param value An object to check.
 * @returns Whether the object is `undefined`, `null`, or has no own properties.
 */
export function isEmptyObject(value?: Record<string, any>): boolean {
  if (value != null) {
    for (const key in value) {
      if (Object.hasOwn(value, key)) return false
    }
  }

  return true
}
