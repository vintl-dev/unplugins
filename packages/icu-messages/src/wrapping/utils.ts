/**
 * Checks whether the plugin wrapping is supported by the provided framework.
 *
 * @param framework Framework name to check in lower case.
 * @returns Whether the framework is one of the supported frameworks.
 */
export function isSupportedLoader(
  framework: string,
): framework is 'rollup' | 'vite' {
  return framework === 'rollup' || framework === 'vite'
}
