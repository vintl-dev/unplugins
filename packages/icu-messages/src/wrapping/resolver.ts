import type { WarningFunction } from '../shared/types.ts'
import { isEmptyObject } from '../utils/object.ts'
import type { WrappingFunctionsMap } from './options.ts'
import { wrapTransform } from './default-wrapper.ts'
import { PluginIneffectiveError } from './errors.ts'

/**
 * A class that holds a map of wrapping functions and exposes a function that
 * can be used to select a single function based on the plugin name.
 */
export class WrappingFunctionSelector<
  PluginType,
  InheritsDefaults extends boolean,
> {
  private readonly wrappers: WrappingFunctionsMap<PluginType>

  /**
   * Constructs a new wrapping functions selector.
   *
   * @param wrappers A map of wrapping functions.
   * @param inheritDefaults Whether to inherit the default wrapping functions.
   * @param onWarn A function that is called when there's a warning.
   */
  constructor(
    wrappers?: WrappingFunctionsMap<PluginType>,
    inheritDefaults?: InheritsDefaults,
    onWarn?: WarningFunction,
  ) {
    const useDefaults = inheritDefaults ?? true

    if (!useDefaults && isEmptyObject(wrappers)) {
      onWarn?.(
        new PluginIneffectiveError(
          'Your configuration does not make use of defaults and does not provide any other wrappers. This transform wrap plugin will be ineffective and probably could be removed.',
        ),
      )
    }

    this.wrappers = {
      ...(useDefaults &&
        ({
          json: wrapTransform,
          'vite:json': wrapTransform,
        } as WrappingFunctionsMap<PluginType>)),
      ...wrappers,
    }
  }

  /**
   * A function that takes in a plugin name and returns a wrapping function for
   * that plugin, if there's any. If there's none, returns `undefined`.
   *
   * It is bound to the object and can be destructured to be used on its own.
   *
   * @param pluginName The plugin name.
   * @returns The wrapping function, or `undefined`.
   */
  public readonly select = (pluginName: string) => {
    return this.wrappers[pluginName]
  }
}
