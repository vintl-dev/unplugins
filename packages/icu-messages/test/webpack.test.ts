import {
  type Configuration as WebpackConfiguration,
  // eslint-disable-next-line import/no-named-default
  default as webpack,
  type StatsCompilation,
} from 'webpack'
import { describe, it, expect, vi } from 'vitest'
import { createFsFromVolume, Volume } from 'memfs'
import { createResolver } from './utils/resolver'
import { basename } from 'pathe'
import type { PluginOptions } from '../dist/webpack.cjs'

Error.stackTraceLimit = 1000

const { icuMessages } = await import('../dist/webpack.cjs')

const resolve = createResolver(import.meta.url)

class WebpackCompilationError extends Error {
  constructor(public causes?: unknown[]) {
    super('Compilation failed')
  }
}

async function buildFile(
  entry: string,
  extendConfig?: (config: WebpackConfiguration) => void | Promise<void>,
) {
  const config: WebpackConfiguration = {
    entry: resolve(entry),
    mode: 'production',
    output: {
      library: { type: 'module' },
      filename: 'index.mjs',
      path: '/dist',
    },
    experiments: { outputModule: true },
    optimization: {
      minimize: false,
      moduleIds: 'named',
      chunkIds: 'named',
      mangleExports: false,
    },
  }

  await extendConfig?.(config)

  const compiler = webpack(config)

  const memfs = createFsFromVolume(new Volume())

  compiler.outputFileSystem = memfs

  await new Promise((resolve, reject) => {
    compiler.run((err, stats) => {
      let resolution: StatsCompilation | undefined
      let rejection: WebpackCompilationError | undefined

      if (err != null) {
        rejection = new WebpackCompilationError([err])
      } else if (stats != null) {
        const info: StatsCompilation = stats.toJson()

        const hasWarnings = stats.hasWarnings()
        const hasErrors = stats.hasErrors()

        if (hasErrors || hasWarnings) {
          if (hasWarnings) {
            console.warn('[webpack] [warn]', info.warnings)
          }

          rejection = new WebpackCompilationError(info.errors)
        } else {
          resolution = stats
        }
      }

      compiler.close((closeErr) => {
        if (closeErr != null) {
          if (rejection == null) {
            rejection = new WebpackCompilationError([closeErr])
          } else {
            ;(rejection.causes ??= []).push(closeErr)
          }
        }

        if (rejection != null) {
          reject(rejection)
          return
        }

        resolve(resolution!)
      })
    })
  })

  return memfs.readFileSync('/dist/index.mjs', { encoding: 'utf-8' })
}

describe(
  'webpack',
  () => {
    it('should generate bundle', async () => {
      const out = await buildFile('fixtures/normal/input.mjs', (config) => {
        ;(config.plugins ??= []).push(
          icuMessages({
            include: '**/*.messages.json',
            format: 'crowdin',
          }),
        )
      })

      expect(out).toMatchSnapshot()
    })

    it('should fail with unresolved formatter', async () => {
      let err: unknown
      try {
        await buildFile('fixtures/normal/input.mjs', (config) => {
          ;(config.plugins ??= []).push(
            icuMessages({
              include: '**/*.messages.json',
              format: 'non-existent',
            }),
          )
        })
      } catch (err_) {
        err = err_
      }

      expect(err).toBeDefined()
    })

    it('should be compatible with other json files', async () => {
      const out = await buildFile('fixtures/with-json/input.mjs', (config) => {
        ;(config.plugins ??= []).push(
          icuMessages({
            include: '**/*.messages.json',
            format: 'crowdin',
          }),
        )
      })

      expect(out).toMatchSnapshot()
    })

    it('should be able to parse using TOML files', async () => {
      const out = await buildFile('fixtures/toml/input.mjs', async (config) => {
        ;(config.plugins ??= []).push(
          icuMessages({
            include: '**/*.messages.toml',
            format: 'crowdin',
            parse: await import('@ltd/j-toml').then(
              (_) => (code: string) => _.parse(code),
            ),
          }),
        )
      })

      expect(out).toMatchSnapshot()
    })

    it('should transform to AST JSON', async () => {
      const out = await buildFile('fixtures/normal/input.mjs', (config) => {
        ;(config.plugins ??= []).push(
          icuMessages({
            include: '**/*.messages.json',
            format: 'crowdin',
            output: {
              format: 'json',
            },
          }),
        )
      })

      expect(out).toMatchSnapshot()
    })

    it('should transform to JSON of raw messages', async () => {
      const out = await buildFile('fixtures/normal/input.mjs', (config) => {
        ;(config.plugins ??= []).push(
          icuMessages({
            include: '**/*.messages.json',
            format: 'crowdin',
            output: {
              type: 'raw',
              format: 'json',
            },
          }),
        )
      })

      expect(out).toMatchSnapshot()
    })

    it('should transform using custom formatter', async () => {
      const out = await buildFile('fixtures/normal/input.mjs', (config) => {
        ;(config.plugins ??= []).push(
          icuMessages({
            include: '**/*.messages.json',
            format: 'crowdin',
            output: {
              format(messages) {
                return JSON.stringify(Object.entries(messages))
              },
            },
          }),
        )
      })

      expect(out).toMatchSnapshot()
    })

    it('should handle errors as defined', async () => {
      const onParseError = vi.fn(function ({ useBuiltinStrategy }) {
        return useBuiltinStrategy('use-message-as-literal')
      } satisfies PluginOptions['onParseError'])

      const out = await buildFile('fixtures/errored/input.mjs', (config) => {
        ;(config.plugins ??= []).push(
          icuMessages({
            format: 'crowdin',
            onParseError,
          }),
        )
      })

      expect(out).toMatchSnapshot()

      expect(onParseError.mock.calls).toHaveLength(1)

      const context = onParseError.mock.calls[0][0]

      expect(context).toBeDefined()

      const { message, messageId, error } = context

      expect({
        message,
        messageId,
        error,
        moduleId: basename(context.moduleId),
        locale: context.parserOptions?.locale?.baseName,
      }).toMatchInlineSnapshot(`
        {
          "error": [SyntaxError: INVALID_TAG],
          "locale": "en",
          "message": "Hello, <bold>{name}</bold!",
          "messageId": "greeting",
          "moduleId": "en.messages.json",
        }
      `)

      expect(onParseError.mock.results[0]).toMatchInlineSnapshot(`
        {
          "type": "return",
          "value": [
            {
              "type": 0,
              "value": "Hello, <bold>{name}</bold!",
            },
          ],
        }
      `)
    })

    it('should pass parsing fc to error handler', async () => {
      const onParseError = vi.fn(function ({
        message,
        useBuiltinStrategy,
        parseMessage,
        parserOptions,
      }) {
        try {
          return parseMessage(message, { ...parserOptions, ignoreTag: true })
        } catch (e) {
          return useBuiltinStrategy('use-message-as-literal')
        }
      } satisfies PluginOptions['onParseError'])

      const out = await buildFile('fixtures/errored/input.mjs', (config) => {
        ;(config.plugins ??= []).push(
          icuMessages({
            format: 'crowdin',
            onParseError,
          }),
        )
      })

      expect(out).toMatchSnapshot()

      const { mock } = onParseError

      expect(mock.calls?.[0]?.[0]?.parseMessage).toBeTypeOf('function')

      expect(mock.results[0]).toMatchInlineSnapshot(`
        {
          "type": "return",
          "value": [
            {
              "type": 0,
              "value": "Hello, <bold>",
            },
            {
              "type": 1,
              "value": "name",
            },
            {
              "type": 0,
              "value": "</bold!",
            },
          ],
        }
      `)
    })
  },
  { timeout: 60_000 },
)
