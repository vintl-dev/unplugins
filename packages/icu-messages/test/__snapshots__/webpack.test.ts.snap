// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`webpack > should be able to parse using TOML files 1`] = `
"/******/ // The require scope
/******/ var __webpack_require__ = {};
/******/ 
/************************************************************************/
/******/ /* webpack/runtime/define property getters */
/******/ (() => {
/******/ 	// define getter functions for harmony exports
/******/ 	__webpack_require__.d = (exports, definition) => {
/******/ 		for(var key in definition) {
/******/ 			if(__webpack_require__.o(definition, key) && !__webpack_require__.o(exports, key)) {
/******/ 				Object.defineProperty(exports, key, { enumerable: true, get: definition[key] });
/******/ 			}
/******/ 		}
/******/ 	};
/******/ })();
/******/ 
/******/ /* webpack/runtime/hasOwnProperty shorthand */
/******/ (() => {
/******/ 	__webpack_require__.o = (obj, prop) => (Object.prototype.hasOwnProperty.call(obj, prop))
/******/ })();
/******/ 
/************************************************************************/
var __webpack_exports__ = {};

// EXPORTS
__webpack_require__.d(__webpack_exports__, {
  "default": () => (/* binding */ example)
});

;// CONCATENATED MODULE: ./test/fixtures/toml/en.messages.toml
const greeting = [
	{
		type: 0,
		value: "Hello, "
	},
	{
		type: 1,
		value: "name"
	},
	{
		type: 0,
		value: "!"
	}
];
/* harmony default export */ const en_messages = ({
	greeting: greeting
});

;// CONCATENATED MODULE: ./test/fixtures/toml/input.mjs


function example() {
  return en_messages.greeting
}

var __webpack_exports__default = __webpack_exports__["default"];
export { __webpack_exports__default as default };
"
`;

exports[`webpack > should be compatible with other json files 1`] = `
"/******/ // The require scope
/******/ var __webpack_require__ = {};
/******/ 
/************************************************************************/
/******/ /* webpack/runtime/define property getters */
/******/ (() => {
/******/ 	// define getter functions for harmony exports
/******/ 	__webpack_require__.d = (exports, definition) => {
/******/ 		for(var key in definition) {
/******/ 			if(__webpack_require__.o(definition, key) && !__webpack_require__.o(exports, key)) {
/******/ 				Object.defineProperty(exports, key, { enumerable: true, get: definition[key] });
/******/ 			}
/******/ 		}
/******/ 	};
/******/ })();
/******/ 
/******/ /* webpack/runtime/hasOwnProperty shorthand */
/******/ (() => {
/******/ 	__webpack_require__.o = (obj, prop) => (Object.prototype.hasOwnProperty.call(obj, prop))
/******/ })();
/******/ 
/************************************************************************/
var __webpack_exports__ = {};

// EXPORTS
__webpack_require__.d(__webpack_exports__, {
  "default": () => (/* binding */ example)
});

;// CONCATENATED MODULE: ./test/fixtures/with-json/options.json
const options_namespaceObject = {"type":0};
;// CONCATENATED MODULE: ./test/fixtures/with-json/en.messages.json
const greeting = [
	{
		type: 0,
		value: "Hello, "
	},
	{
		type: 1,
		value: "name"
	},
	{
		type: 0,
		value: "!"
	}
];
const goodbye = [
	{
		type: 0,
		value: "Farewell, "
	},
	{
		type: 1,
		value: "name"
	},
	{
		type: 0,
		value: "!"
	}
];
/* harmony default export */ const en_messages = ({
	greeting: greeting,
	goodbye: goodbye
});

;// CONCATENATED MODULE: ./test/fixtures/with-json/input.mjs



const messageTypes = [
  [0, 'greeting'],
  [1, 'goodbye'],
]

function example() {
  const messageId = messageTypes.find(([type]) => options_namespaceObject.type === type)
  if (messageId == null) throw new Error(\`Unknown type \${options_namespaceObject.type}\`)
  return en_messages[messageId]
}

var __webpack_exports__default = __webpack_exports__["default"];
export { __webpack_exports__default as default };
"
`;

exports[`webpack > should generate bundle 1`] = `
"/******/ // The require scope
/******/ var __webpack_require__ = {};
/******/ 
/************************************************************************/
/******/ /* webpack/runtime/define property getters */
/******/ (() => {
/******/ 	// define getter functions for harmony exports
/******/ 	__webpack_require__.d = (exports, definition) => {
/******/ 		for(var key in definition) {
/******/ 			if(__webpack_require__.o(definition, key) && !__webpack_require__.o(exports, key)) {
/******/ 				Object.defineProperty(exports, key, { enumerable: true, get: definition[key] });
/******/ 			}
/******/ 		}
/******/ 	};
/******/ })();
/******/ 
/******/ /* webpack/runtime/hasOwnProperty shorthand */
/******/ (() => {
/******/ 	__webpack_require__.o = (obj, prop) => (Object.prototype.hasOwnProperty.call(obj, prop))
/******/ })();
/******/ 
/************************************************************************/
var __webpack_exports__ = {};

// EXPORTS
__webpack_require__.d(__webpack_exports__, {
  "default": () => (/* binding */ example)
});

;// CONCATENATED MODULE: ./test/fixtures/normal/en.messages.json
const greeting = [
	{
		type: 0,
		value: "Hello, "
	},
	{
		type: 1,
		value: "name"
	},
	{
		type: 0,
		value: "!"
	}
];
/* harmony default export */ const en_messages = ({
	greeting: greeting
});

;// CONCATENATED MODULE: ./test/fixtures/normal/input.mjs


function example() {
  return en_messages
}

var __webpack_exports__default = __webpack_exports__["default"];
export { __webpack_exports__default as default };
"
`;

exports[`webpack > should handle errors as defined 1`] = `
"/******/ // The require scope
/******/ var __webpack_require__ = {};
/******/ 
/************************************************************************/
/******/ /* webpack/runtime/define property getters */
/******/ (() => {
/******/ 	// define getter functions for harmony exports
/******/ 	__webpack_require__.d = (exports, definition) => {
/******/ 		for(var key in definition) {
/******/ 			if(__webpack_require__.o(definition, key) && !__webpack_require__.o(exports, key)) {
/******/ 				Object.defineProperty(exports, key, { enumerable: true, get: definition[key] });
/******/ 			}
/******/ 		}
/******/ 	};
/******/ })();
/******/ 
/******/ /* webpack/runtime/hasOwnProperty shorthand */
/******/ (() => {
/******/ 	__webpack_require__.o = (obj, prop) => (Object.prototype.hasOwnProperty.call(obj, prop))
/******/ })();
/******/ 
/************************************************************************/
var __webpack_exports__ = {};

// EXPORTS
__webpack_require__.d(__webpack_exports__, {
  "default": () => (/* binding */ example)
});

;// CONCATENATED MODULE: ./test/fixtures/errored/en.messages.json
const greeting = [
	{
		type: 0,
		value: "Hello, <bold>{name}</bold!"
	}
];
/* harmony default export */ const en_messages = ({
	greeting: greeting
});

;// CONCATENATED MODULE: ./test/fixtures/errored/input.mjs


function example() {
  return en_messages
}

var __webpack_exports__default = __webpack_exports__["default"];
export { __webpack_exports__default as default };
"
`;

exports[`webpack > should pass parsing fc to error handler 1`] = `
"/******/ // The require scope
/******/ var __webpack_require__ = {};
/******/ 
/************************************************************************/
/******/ /* webpack/runtime/define property getters */
/******/ (() => {
/******/ 	// define getter functions for harmony exports
/******/ 	__webpack_require__.d = (exports, definition) => {
/******/ 		for(var key in definition) {
/******/ 			if(__webpack_require__.o(definition, key) && !__webpack_require__.o(exports, key)) {
/******/ 				Object.defineProperty(exports, key, { enumerable: true, get: definition[key] });
/******/ 			}
/******/ 		}
/******/ 	};
/******/ })();
/******/ 
/******/ /* webpack/runtime/hasOwnProperty shorthand */
/******/ (() => {
/******/ 	__webpack_require__.o = (obj, prop) => (Object.prototype.hasOwnProperty.call(obj, prop))
/******/ })();
/******/ 
/************************************************************************/
var __webpack_exports__ = {};

// EXPORTS
__webpack_require__.d(__webpack_exports__, {
  "default": () => (/* binding */ example)
});

;// CONCATENATED MODULE: ./test/fixtures/errored/en.messages.json
const greeting = [
	{
		type: 0,
		value: "Hello, <bold>"
	},
	{
		type: 1,
		value: "name"
	},
	{
		type: 0,
		value: "</bold!"
	}
];
/* harmony default export */ const en_messages = ({
	greeting: greeting
});

;// CONCATENATED MODULE: ./test/fixtures/errored/input.mjs


function example() {
  return en_messages
}

var __webpack_exports__default = __webpack_exports__["default"];
export { __webpack_exports__default as default };
"
`;

exports[`webpack > should transform to AST JSON 1`] = `
"/******/ // The require scope
/******/ var __webpack_require__ = {};
/******/ 
/************************************************************************/
/******/ /* webpack/runtime/define property getters */
/******/ (() => {
/******/ 	// define getter functions for harmony exports
/******/ 	__webpack_require__.d = (exports, definition) => {
/******/ 		for(var key in definition) {
/******/ 			if(__webpack_require__.o(definition, key) && !__webpack_require__.o(exports, key)) {
/******/ 				Object.defineProperty(exports, key, { enumerable: true, get: definition[key] });
/******/ 			}
/******/ 		}
/******/ 	};
/******/ })();
/******/ 
/******/ /* webpack/runtime/hasOwnProperty shorthand */
/******/ (() => {
/******/ 	__webpack_require__.o = (obj, prop) => (Object.prototype.hasOwnProperty.call(obj, prop))
/******/ })();
/******/ 
/************************************************************************/
var __webpack_exports__ = {};

// EXPORTS
__webpack_require__.d(__webpack_exports__, {
  "default": () => (/* binding */ example)
});

;// CONCATENATED MODULE: ./test/fixtures/normal/en.messages.json
const en_messages_namespaceObject = /*#__PURE__*/JSON.parse('{"greeting":[{"type":0,"value":"Hello, "},{"type":1,"value":"name"},{"type":0,"value":"!"}]}');
;// CONCATENATED MODULE: ./test/fixtures/normal/input.mjs


function example() {
  return en_messages_namespaceObject
}

var __webpack_exports__default = __webpack_exports__["default"];
export { __webpack_exports__default as default };
"
`;

exports[`webpack > should transform to JSON of raw messages 1`] = `
"/******/ // The require scope
/******/ var __webpack_require__ = {};
/******/ 
/************************************************************************/
/******/ /* webpack/runtime/define property getters */
/******/ (() => {
/******/ 	// define getter functions for harmony exports
/******/ 	__webpack_require__.d = (exports, definition) => {
/******/ 		for(var key in definition) {
/******/ 			if(__webpack_require__.o(definition, key) && !__webpack_require__.o(exports, key)) {
/******/ 				Object.defineProperty(exports, key, { enumerable: true, get: definition[key] });
/******/ 			}
/******/ 		}
/******/ 	};
/******/ })();
/******/ 
/******/ /* webpack/runtime/hasOwnProperty shorthand */
/******/ (() => {
/******/ 	__webpack_require__.o = (obj, prop) => (Object.prototype.hasOwnProperty.call(obj, prop))
/******/ })();
/******/ 
/************************************************************************/
var __webpack_exports__ = {};

// EXPORTS
__webpack_require__.d(__webpack_exports__, {
  "default": () => (/* binding */ example)
});

;// CONCATENATED MODULE: ./test/fixtures/normal/en.messages.json
const en_messages_namespaceObject = /*#__PURE__*/JSON.parse('{"greeting":"Hello, {name}!"}');
;// CONCATENATED MODULE: ./test/fixtures/normal/input.mjs


function example() {
  return en_messages_namespaceObject
}

var __webpack_exports__default = __webpack_exports__["default"];
export { __webpack_exports__default as default };
"
`;

exports[`webpack > should transform using custom formatter 1`] = `
"/******/ var __webpack_modules__ = ({

/***/ "./test/fixtures/normal/en.messages.json":
/***/ (() => {

[["greeting",[{"type":0,"value":"Hello, "},{"type":1,"value":"name"},{"type":0,"value":"!"}]]]

/***/ })

/******/ });
/************************************************************************/
/******/ // The module cache
/******/ var __webpack_module_cache__ = {};
/******/ 
/******/ // The require function
/******/ function __webpack_require__(moduleId) {
/******/ 	// Check if module is in cache
/******/ 	var cachedModule = __webpack_module_cache__[moduleId];
/******/ 	if (cachedModule !== undefined) {
/******/ 		return cachedModule.exports;
/******/ 	}
/******/ 	// Create a new module (and put it into the cache)
/******/ 	var module = __webpack_module_cache__[moduleId] = {
/******/ 		// no module.id needed
/******/ 		// no module.loaded needed
/******/ 		exports: {}
/******/ 	};
/******/ 
/******/ 	// Execute the module function
/******/ 	__webpack_modules__[moduleId](module, module.exports, __webpack_require__);
/******/ 
/******/ 	// Return the exports of the module
/******/ 	return module.exports;
/******/ }
/******/ 
/************************************************************************/
/******/ /* webpack/runtime/define property getters */
/******/ (() => {
/******/ 	// define getter functions for harmony exports
/******/ 	__webpack_require__.d = (exports, definition) => {
/******/ 		for(var key in definition) {
/******/ 			if(__webpack_require__.o(definition, key) && !__webpack_require__.o(exports, key)) {
/******/ 				Object.defineProperty(exports, key, { enumerable: true, get: definition[key] });
/******/ 			}
/******/ 		}
/******/ 	};
/******/ })();
/******/ 
/******/ /* webpack/runtime/hasOwnProperty shorthand */
/******/ (() => {
/******/ 	__webpack_require__.o = (obj, prop) => (Object.prototype.hasOwnProperty.call(obj, prop))
/******/ })();
/******/ 
/************************************************************************/
var __webpack_exports__ = {};
// This entry need to be wrapped in an IIFE because it need to be isolated against other modules in the chunk.
(() => {
/* harmony export */ __webpack_require__.d(__webpack_exports__, {
/* harmony export */   "default": () => (/* binding */ example)
/* harmony export */ });
/* harmony import */ var _en_messages_json__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__("./test/fixtures/normal/en.messages.json");


function example() {
  return _en_messages_json__WEBPACK_IMPORTED_MODULE_0__
}

})();

var __webpack_exports__default = __webpack_exports__["default"];
export { __webpack_exports__default as default };
"
`;
